libgadget/blackhole.c

```diff
@@ -336,6 +472,22 @@ blackhole(const ActiveParticles * act, ForceTree * tree, FILE * FdBlackHoles, FI
     tw_feedback->result_type_elsize = sizeof(TreeWalkResultBHFeedback);
     tw_feedback->tree = tree;
     tw_feedback->priv = priv;
+    
+    
+    /*************************************************************************/
+    /*  Dynamical Friction Treewalk */
+
+    MPIU_Barrier(MPI_COMM_WORLD);
+    message(0, "Beginning dynamical friction calculation. \n");
+    /* Environment variables for DF */
+    priv->BH_SurroundingRmsVel = mymalloc("BH_SurroundingRmsVel", SlotsManager->info[5].size * sizeof(priv->BH_SurroundingRmsVel));
+    priv->BH_SurroundingVel = (MyFloat (*) [3]) mymalloc("BH_SurroundingVel", 3* SlotsManager->info[5].size * sizeof(priv->BH_SurroundingVel[0]));
+    priv->BH_SurroundingParticles = mymalloc("BH_SurroundingParticles", SlotsManager->info[5].size * sizeof(priv->BH_SurroundingParticles));
+    priv->BH_SurroundingDensity = mymalloc("BH_SurroundingDensity", SlotsManager->info[5].size * sizeof(priv->BH_SurroundingDensity));
+
+    treewalk_run(tw_dynfric, act->ActiveParticle, act->NumActiveParticle);
```

Comment:
Guard this by if(DynFrictionMethod > 0)

Reply:
I tried to guard this by
if(DynFrictionMethod > 0)
         treewalk_run(tw_dynfric, act->ActiveParticle, act->NumActiveParticle);

but that would make the run with DynFrictionMethod = 0 stuck after the first time step, at 
[ 000109.26 ] Attempting a domain exchange

Reply:
hmm, could you figure out why that is and fix it? My guess is that it is because one of the *Accel variables is not initialised in this case, so the particle receives an arbitrary kick and moves out of the box.